src/platformAccessory.ts

```diff
@@ -28,10 +28,25 @@ export const enum PlatformAccessoryEvent {
   IDENTIFY = "identify",
 }
 
-export declare interface PlatformAccessory {
+export interface PlatformAccessory {
 
   on(event: "identify", listener: () => void): this;
+  on(event: "identify", listener: (paired: boolean) => void): this;
 
+  /**
+   * @deprecated This version with the noopCallback will be removed in a future version
+   */
+  on(event: "identify", listener: (paired: boolean, noopCallback: () => void) => void): this;
+
+  /**
+   * @deprecated Kept for back compat, do not rely on paired or callback.
+   */
+  emit(event: "identify", paired: boolean, callback: () => void): boolean;
+
+  /**
+   * @alpha This is the future supported version of emit("identify"), invalid until
+   *        the next breaking change.
+   */
   emit(event: "identify"): boolean;
```

Comment:
This is the desired reality but if someone were to emit like this today, it would cause bugs.